tests/components/min_max/test_sensor.py

```diff
@@ -14,329 +13,334 @@
+    config = {
+        "sensor": {
+            "platform": "min_max",
+            "name": "test_max",
+            "type": "max",
+            "entity_ids": ["sensor.test_1", "sensor.test_2", "sensor.test_3"],
         }
+    }
 
-        assert setup_component(self.hass, "sensor", config)
+    assert await async_setup_component(hass, "sensor", config)
+    await hass.async_block_till_done()
 
-        entity_ids = config["sensor"]["entity_ids"]
+    entity_ids = config["sensor"]["entity_ids"]
 
-        for entity_id, value in dict(zip(entity_ids, self.values)).items():
-            self.hass.states.set(entity_id, value)
-            self.hass.block_till_done()
+    for entity_id, value in dict(zip(entity_ids, VALUES)).items():
+        hass.states.async_set(entity_id, value)
+        await hass.async_block_till_done()
 
-        state = self.hass.states.get("sensor.test_max")
+    state = hass.states.get("sensor.test_max")
 
-        assert str(float(self.max)) == state.state
-        assert entity_ids[2] == state.attributes.get("min_entity_id")
-        assert self.min == state.attributes.get("min_value")
-        assert entity_ids[1] == state.attributes.get("max_entity_id")
-        assert self.mean == state.attributes.get("mean")
-        assert self.median == state.attributes.get("median")
+    assert str(float(MAX_VALUE)) == state.state
+    assert entity_ids[2] == state.attributes.get("min_entity_id")
+    assert MIN_VALUE == state.attributes.get("min_value")
+    assert entity_ids[1] == state.attributes.get("max_entity_id")
+    assert MEAN == state.attributes.get("mean")
+    assert MEDIAN == state.attributes.get("median")
 
-    def test_mean_sensor(self):
-        """Test the mean sensor."""
-        config = {
-            "sensor": {
-                "platform": "min_max",
-                "name": "test_mean",
-                "type": "mean",
-                "entity_ids": ["sensor.test_1", "sensor.test_2", "sensor.test_3"],
-            }
+
+async def test_mean_sensor(hass):
+    """Test the mean sensor."""
+    config = {
+        "sensor": {
+            "platform": "min_max",
+            "name": "test_mean",
+            "type": "mean",
+            "entity_ids": ["sensor.test_1", "sensor.test_2", "sensor.test_3"],
         }
+    }
+
+    assert await async_setup_component(hass, "sensor", config)
+    await hass.async_block_till_done()
 
-        assert setup_component(self.hass, "sensor", config)
+    entity_ids = config["sensor"]["entity_ids"]
 
-        entity_ids = config["sensor"]["entity_ids"]
+    for entity_id, value in dict(zip(entity_ids, VALUES)).items():
+        hass.states.async_set(entity_id, value)
+        await hass.async_block_till_done()
 
-        for entity_id, value in dict(zip(entity_ids, self.values)).items():
-            self.hass.states.set(entity_id, value)
-            self.hass.block_till_done()
+    state = hass.states.get("sensor.test_mean")
+
+    assert str(float(MEAN)) == state.state
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6285789)